From 8bea05047475cee1b5476742507b310a663ac59d Mon Sep 17 00:00:00 2001
From: Arun Ravindran <arun.ravindran@intel.com>
Date: Tue, 30 Apr 2013 16:59:15 +0300
Subject: [PORT FROM MAIN] Telephony: STK App not seen after SIM Hot swap

BZ: 78983

The root cause of this issue is that when the SIM is removed the
UiccCard.java in its Update() function, discards the CatService,
and sets the internal object to NULL. When the SIM is re inserted
a new CatService instance is created only when already existing
singleton is null, and hence returns the disposed obbject, which
has de registered for notification for STK PROACTIVE COMMANDS.
This results in STK app disappearing for ever after SIM hot Swap.

This patch fixes the issue by removing the call to dispose() of
CatService from UiccCard.java.

Change-Id: I77cf98f2249981b95954edfc7d19d91878018c1c
Orig-Change-Id: I0313642e7886527daee87477fc8cae10efee02f4
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../android/internal/telephony/uicc/UiccCard.java  |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/src/java/com/android/internal/telephony/uicc/UiccCard.java b/src/java/com/android/internal/telephony/uicc/UiccCard.java
index 7d63522..c904180 100644
--- a/src/java/com/android/internal/telephony/uicc/UiccCard.java
+++ b/src/java/com/android/internal/telephony/uicc/UiccCard.java
@@ -137,9 +137,7 @@ public class UiccCard {
                                                      mContext,
                                                      this);
             } else {
-                if (mCatService != null) {
-                    mCatService.dispose();
-                }
+                // Just set the member variable to null.
                 mCatService = null;
             }
 
-- 
1.7.4.1

